### PR TITLE
Fixed major issues fixes #15

### LINKED ISSUE #15: Support for Kobotoolbox?
A lot of organisations use the free institutional accounts sponsored by UN-OCHA of the Kobotoolbox:
https://www.kobotoolbox.org/

I believe Kobotoolbox is an Ona.io fork but significantly diverged. I tried connecting to it via QRealTime, but it didn't work. Help apprechiated!

### PR DESCRIPTION
**Pull request by sawanrawat**
Fixed "Import" and "Make online" and removed redundant lines of code fixes #15

### CODE REVIEW

File: QRealTime_dialog.py [hunk 1]
```diff
@@ -33,7 +33,10 @@
 from six.moves import range
 from qgis.core import QgsMessageLog, Qgis
 import csv
+import json
+import ast
 import codecs
```
Comment (shivareddyiirs) [inline QRealTime_dialog.py]: does codecs is used anywhere?

File: QRealTime.py [hunk 2]
```diff
@@ -361,22 +348,22 @@ def updateLayer(self,layer,xml):
...
-            print ('first attribute'+ fieldName)
+#            print ('first attribute'+ fieldName)
             inputs=root[1].findall('.//*[@ref]')
             if fieldType[:3]!='geo':
-                print('creating new field:'+ fieldName)
+                #print('creating new field:'+ fieldName)
                 isHidden= True
                 for input in inputs:
-                    if fieldName == input.attrib['ref'].split('/')[-1]:
+                    if fieldName == input.attrib['ref'].split('/')[-1].replace("_", " "):
```
Comment (shivareddyiirs) [inline QRealTime.py:365]: did you add _try catch_ block to check if any attribute name  has  '_'?

File: QRealTime.py [hunk 3]
```diff
@@ -41,26 +42,11 @@
 import xml.etree.ElementTree as ET
 import subprocess
```
Comment (shivareddyiirs) [inline QRealTime.py]: this import  can also be removed just check

**Review by shivareddyiirs — CHANGES_REQUESTED** [review]
Minor changes to remove the some extra import and add try catch for checking attribute name having '_'

File: QRealTime_dialog.py [hunk 4]
```diff
@@ -195,14 +201,24 @@ def sendForm(self,xForm_id,xForm):
         if form_key:
```
Comment (shivareddyiirs) [inline QRealTime_dialog.py]: This condition was kept to check if form already exist with given name. 
Did you check supported methods like 'put' for updating the existing form?

Comment (sawanrawat) [reply QRealTime_dialog.py]: Sir, I have made changes to check if the form already exists with given name and updating it using a PATCH request in the new commit. The sync option also works now.

File: QRealTime_dialog.py [hunk 5]
```diff
@@ -369,44 +381,86 @@ def transformToLayerSRS(self, pPoint):
...
+            for form in responseJSON:
+                if str(form['title'])==XFormKey:
+                    formID=str(form['formid'])
+            urlData='https://kc.humanitarianresponse.info/api/v1/data/'+formID
+            responseData = requests.get(urlData,auth=(self.getValue('user'),self.getValue('password')),verify=False)
+            data=json.loads(responseData.text)
+            uuidList = self.getUUIDList(layer)
+            for submission in data:
+                if submission['meta/instanceID'] not in uuidList:
+                    xmlurl="https://kobo.humanitarianresponse.info/assets/"+submission['_xform_id_string']
```
Comment (shivareddyiirs) [inline QRealTime_dialog.py]: url is not changing in the loop.  i think '_xform_id_string' should be replaced with a variable which iterates through all instance ids.

File: QRealTime_dialog.py [hunk 6]
```diff
@@ -369,44 +381,86 @@ def transformToLayerSRS(self, pPoint):
...
+                    para={'format':'xml'}
+                    response2=requests.get(xmlurl,auth=(self.getValue('user'),self.getValue('password')),params=para)
```
Comment (shivareddyiirs) [inline QRealTime_dialog.py]: also nowhere lastID is used.

**Review by shivareddyiirs — CHANGES_REQUESTED** [review]
some changes needed for synch to work propertly

**Review by shivareddyiirs — COMMENTED** [review]
What if form with same name already exist? I think we should allow form to be updated by the user

**Comment by sawanrawat** [comment]
the sendForm function in QRealTime_dialog.py does check if the form exists.
I tried sending patch and post requests to the asset endpoints the the form
isn't getting updated. I'll look into it further.

File: QRealTime_dialog.py [hunk 7]
```diff
@@ -263,35 +293,36 @@ def updateLayer(self,layer,dataDict,geoField):
...
+                    print('dict is',odkFeature)
+                    wktGeom = self.guessWKTGeomType(odkFeature[geoField])
+                    print ("WKTGeom is",wktGeom)
+                    if wktGeom[:3] != layerGeo[:3]:
+                        continue
+                    qgisGeom = QgsGeometry.fromWkt(wktGeom)
+                    print('geom is',qgisGeom)
+                    qgisFeature.setGeometry(qgisGeom)
+                    qgisFeature.initAttributes(len(QgisFieldsList))
+                    for fieldName, fieldValue in six.iteritems(odkFeature):
```
Comment (shivareddyiirs) [inline QRealTime_dialog.py:315]: i think there is no need to use six.iteritems

Comment (sawanrawat) [reply QRealTime_dialog.py:315]: yes sir, the same result can be obtained by using odkFeature.items() I'll make the necessary changes.

**Comment by shivareddyiirs** [comment]
fixes #15

**Pull request merged**